component/componentattribute/logger.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package componentattribute // import "go.opentelemetry.io/collector/component/componentattribute"
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var _ zapcore.Core = (*coreWithout)(nil)
+
+type coreWithout struct {
+	zapcore.Core
+	from  *zap.Logger
+	attrs attribute.Set
+}
+
+func NewLogger(from *zap.Logger, attrs *attribute.Set) *zap.Logger {
```

Comment:
I'm not sure how to avoid exposing this. It's used in component tests, and also in the graph when instantiating the loggers for components.